x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -163,6 +167,28 @@ public LocalExecutionPlan plan(PhysicalPlan node) {
             AggregateExec.class,
             a -> a.getMode() == AggregateExec.Mode.FINAL ? new ProjectExec(a.source(), a, Expressions.asAttributes(a.aggregates())) : a
         );
+        if (configuration.pragmas().timeSeriesMode()) {
+            var tsidAttribute = new FieldAttribute(Source.EMPTY, "_tsid", new EsField("_tsid", DataTypes.KEYWORD, Map.of(), true));
```

Comment:
@martijnvg can you please explain what is the high-level purpose for this change here? From other changes in this PR I get that `_tsid` and `@timestamp` are necessary/specific fields related to time series?
Are they similar to https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java#L32? (an internal "field" that is not visible to users in any way, that we clean up properly, but we do need it)

Reply:
This is meant as internal fields to be used by other operators (e.g. time series grouping operator). Users shouldn't see the `_tsid` and `@timestamp` fields. (unless specifically asking for `@timestamp`).
Initially I was thinking of making this visible to the user, but the `_tsid` used to contain the user readable time series id, but today it is a hash (not useful for users).